### PR TITLE
[Clang][LLVM][SVE2.1]  Created intrinsics for DUPQ instr.

### PR DESCRIPTION
**Pull request by Lukacma**
This patch adds clang and llvm support for following intrinsic and maps it to DUPQ instruction:
```
   // Variants are also available for:
   // _s8, _u16, _s16, _u32, _s32, _u64, _s64
   // _bf16, _f16, _f32, _f64
   svuint8_t svdup_laneq[_u8](svuint8_t zn, uint64_t imm_idx);
```

### CODE REVIEW

File: llvm/lib/Target/AArch64/SVEInstrFormats.td [hunk 1]
```diff
@@ -10007,6 +10007,16 @@ multiclass sve2p1_dupq<string mnemonic> {
     bits<1> index;
     let Inst{20} = index;
   }
+
+  def : SVE_2_Op_Imm_Pat<nxv16i8, Op, nxv16i8, i32, VectorIndexB32b_timm, !cast<Instruction>(NAME # _B)>;
```
Comment (CarolineConcatto) [inline llvm/lib/Target/AArch64/SVEInstrFormats.td:10011]: Can you change the vector index in lines 9994, 9998, 10002, 10006?

File: clang/include/clang/Basic/arm_sve.td [hunk 2]
```diff
@@ -2215,6 +2215,15 @@ let TargetGuard = "sve2p1" in {
   def SVTBXQ : SInst<"svtbxq[_{d}]", "dddu", "cUcsUsiUilUlbhfd", MergeNone, "aarch64_sve_tbxq">;
   // EXTQ
   def EXTQ : SInst<"svextq[_{d}]", "dddk", "cUcsUsiUilUlbhfd", MergeNone, "aarch64_sve_extq", [], [ImmCheck<2, ImmCheck0_15>]>;
+  // DUPQ
+  def SVDUP_LANEQ_B  : SInst<"svdup_laneq[_{d}]", "ddi",  "cUc", MergeNone, "aarch64_sve_dup_laneq", [IsStreamingCompatible], [ImmCheck<1, ImmCheck0_15>]>;
```
Comment (CarolineConcatto) [inline clang/include/clang/Basic/arm_sve.td:2219]: We should add test in Sema for the immediate in:
clang/test/Sema/aarch64-sve2p1-intrinsics/acle_sve2p1_imm.cpp

**Review by CarolineConcatto — COMMENTED** [review]
I would change the   commit message to have:
[Clang][LLVM][SVE2.1]  Created intrinsics for DUPQ instr

File: llvm/lib/Target/AArch64/SVEInstrFormats.td (hunk 1 above)
Comment (Lukacma) [reply llvm/lib/Target/AArch64/SVEInstrFormats.td:10011]: You mean changing them to 64 bits ones?

Comment (momchil-velikov) [reply llvm/lib/Target/AArch64/SVEInstrFormats.td:10011]: Change them to `_timm`.

Comment (Lukacma) [reply llvm/lib/Target/AArch64/SVEInstrFormats.td:10011]: DOne

File: clang/include/clang/Basic/arm_sve.td (hunk 2 above)
Comment (Lukacma) [reply clang/include/clang/Basic/arm_sve.td:2219]: Done

**Review by CarolineConcatto — APPROVED** [review]

**Pull request merged**